tasks/modules/install-modules.yml

```diff
@@ -8,9 +8,11 @@
     - nginx_install_epel_release | bool
 
 - name: (Amazon Linux/RHEL 7) Install GeoIP2 dependency
+  vars:
+    distribution: "{{ ansible_facts['distribution'] }} {{ ansible_facts['system'] }} {{ ansible_facts['distribution_version'] }}"
   when:
-    - (ansible_facts['distribution'] == "RedHat" and ansible_facts['distribution_major_version'] is version('7', '=='))
-      or ansible_facts['distribution'] == "Amazon"
+    - (ansible_facts['distribution'] == "RedHat" and ansible_facts['distribution_major_version'] is version('7', '==')) or ansible_facts['distribution'] == "Amazon
+      and distribution != 'Amazon LInux 2023'"
```

Comment:
While at it, if we are going to separate this conditional into two lines, let's do it at the `or`, not like it is done right now.